### PR TITLE
[fix](orc) check all the cases before build_search_argument

### PR DESCRIPTION
**Pull request by suxiaogang223**
### What problem does this PR solve?
In the old logic, the `check_expr_can_push_down` function does not check whether the `orc::Literal` are constructed successfully, but only checks during `build_search_argument`. However, if it is found that the `orc::Literal` fails to be constructed after `builder->startNot`, it will fail because the builder cannot end `startNot`.
Therefore, we advance the behavior of constructing `orc::Literal` to the `check_expr_can_push_down` function and save the result to the map, so that it will never fail in the `build_search_argument` phase.

Related PR: #43255

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [x] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by suxiaogang223** [comment]
run buildall

**Comment by suxiaogang223** [comment]
run buildall

**Comment by suxiaogang223** [comment]
run buildall

**Comment by suxiaogang223** [comment]
run external

**Comment by suxiaogang223** [comment]
run buildall

File: be/src/vec/exec/format/orc/vorc_reader.cpp [hunk 1]
```diff
@@ -624,15 +625,37 @@ std::tuple<bool, orc::Literal, orc::PredicateDataType> OrcReader::_make_orc_lite
...
+// check if the slot of expr can be pushed down to orc reader and make orc predicate type
 bool OrcReader::_check_slot_can_push_down(const VExprSPtr& expr) {
     if (!expr->children()[0]->is_slot_ref()) {
         return false;
     }
     const auto* slot_ref = static_cast<const VSlotRef*>(expr->children()[0].get());
     // check if the slot exists in orc file and not partition column
-    return _col_name_to_file_col_name.contains(slot_ref->expr_name()) &&
-           !_lazy_read_ctx.predicate_partition_columns.contains(slot_ref->expr_name());
+    if (!_col_name_to_file_col_name.contains(slot_ref->expr_name()) &&
```
Comment (morningman) [inline be/src/vec/exec/format/orc/vorc_reader.cpp:635]:
```suggestion
    if (!_col_name_to_file_col_name.contains(slot_ref->expr_name()) ||
```

**Comment by suxiaogang223** [comment]
run buildall

**Review by morningman — APPROVED** [review]

**Review by kaka11chen — APPROVED** [review]
LGTM

**Pull request merged**